src/sage/graphs/graph_decompositions/slice_decomposition.pyx

```diff
@@ -0,0 +1,970 @@
+            sage: SD.lexBFS_order()
+            [0, 1, 4, 5, 2, 6, 3, 9, 7, 8]
+            sage: _is_valid_lex_BFS_order(G, SD.lexBFS_order())
+            True
+
+        TESTS:
+
+            sage: from sage.graphs.traversals import _is_valid_lex_BFS_order
+            sage: for _ in range(5):
+            ....:   G = graphs.RandomGNP(15, 0.3)
+            ....:   SD = G.slice_decomposition()
+            ....:   _is_valid_lex_BFS_order(G, SD.lexBFS_order())
+            True
+            True
+            True
+            True
+            True
+
+        """
+        return list(self.sigma)
+
+    def xslice_data(self, v):
+        r"""
+        Return the data about the x-slice of the vertex `v`.
+
+        This method is a wrapper around :meth:`SliceDecomposition.__getitem__`
+
+        TESTS:
+
+            sage: G = graphs.RandomGNP(15, 0.3)
+            sage: SD = G.slice_decomposition()
+            sage: all(SD[v] == SD.xslice_data(v) for v in G)
+            True
+
+        """
+        return self[v]
+
+    def slice (self, v):
+        r"""
+        Return the slice of the vertex `v`.
+
+        The slice of `v` is the list of vertices `u` such that the neighbors of
+        `u` that are before `v` in the lexBFS order are that same that the
+        neighbors of `v` that are before `v` in the lexBFS order (*i.e.*, the
+        lexicographic label of `v`). It can be shown that it is a factor of the
+        lexBFS order.
+
+        INPUT:
+
+        - ``v`` -- a vertex of the graph corresponding to the slice
+          decomposition.
+
+        OUTPUT:
+
+        A list of vertices
+
+        EXAMPLES:
+
+        ::
+
+            sage: G = Graph('L~mpn~Nrv{^o~_').relabel('abcdefguvwxyz',inplace=False)
+            sage: SD = G.slice_decomposition(initial_vertex='x')
+            sage: SD.slice('a')
+            ['a', 'b', 'c', 'd', 'e', 'f', 'g']
+
+        The vertices of the slice have the same neighborhood "on the left"::
+
+            sage: pos = lambda v: SD.lexBFS_order().index(v)
+            sage: lla = set(SD.lexicographic_label('a'))
+            sage: all(lla == {u for u in G.neighbors(v) if pos(u) < pos('a')} \
+            ....:       for v in SD.slice('a'))
+            True
+
+        The slice is a factor of the lexBFS order::
+
+            sage: ''.join(SD.slice('a')) in ''.join(SD.lexBFS_order())
+            True
+
+        The slice of the initial vertex is the whole graph::
+
+            sage: SD.slice('x') == SD.lexBFS_order()
+            True
+
+        TESTS:
+
+            sage: SD.slice('Michael')
+            Traceback (most recent call last):
+            ...
+            LookupError: vertex (Michael) does not appear in the slice decomposition
+        """
+        if v not in self.sigma_inv:
+            raise LookupError(f"vertex ({v}) does not appear in the slice "
+                              "decomposition")
+        i = self.sigma_inv[v]
```

Comment:
`cdef int i = ` ?

Reply:
done in f3d502e0a17cc39325d09f1d4a00c60dd3aa278e